src/tests/tests.rs

```diff
@@ -0,0 +1,14 @@
+use super::Histogram;
+
+#[path = "helpers.rs"]
```

Comment:
Hmmm... Having the same motivations, I come to the opposite conclusion. :( I want branches that represent a feature in its entirety, not a semi-arbitrary fragment of an overall feature. I would like to be able to locate the merge that introduced a bug, and then bisect along the branch that led to that merge, knowing that it has a good chance of navigating through every important change that's part of that feature. In what way is #18 not necessary for #16? It was merged before it, and #16 would not apply cleanly without it. And yet, the motivation for #18 is absent until you get to #16: the addition of another large block of tests.

IMO, developing an abstraction (of which splitting unit tests is a simple example) in a vacuum is a good way to get an abstraction that doesn't fit with the end goal. Or, going the other way, retroactively splitting it out is a good way to end up with a mangled branch -- on my first attempt, I didn't delete the old `tests.rs`, for instance. 

Oh well. I guess this isn't likely to change your mind. We all come at this with different histories and biases.